src/appleseed/renderer/kernel/intersection/embreescene.cpp

```diff
@@ -448,12 +448,47 @@ void EmbreeScene::intersect(ShadingPoint& shading_point) const
         const uint32 v1_idx = geometry_data->m_primitives[rayhit.hit.primID * 3 + 1];
         const uint32 v2_idx = geometry_data->m_primitives[rayhit.hit.primID * 3 + 2];
 
-        const TriangleType triangle(
-            Vector3d(geometry_data->m_vertices[v0_idx]),
-            Vector3d(geometry_data->m_vertices[v1_idx]),
-            Vector3d(geometry_data->m_vertices[v2_idx]));
+        if (geometry_data->m_motion_steps_count > 1)
+        {
+            // N is the last motion step index.
+            const uint32 N = geometry_data->m_motion_steps_count - 1;
```

Comment:
Spell variables in lower case. May I suggest `last_motion_step_index` instead?

Reply:
Ah, no prob. It was just a variable name from my paper scribbling.